### DISCUSSION TITLE
Why unable to store multiple records using createMany in Prisma?

### DISCUSSION THREAD

**Discussion by pawan-kr1997** [prisma/prisma#15807] [Q&A] [answered]
I am trying to create multiple records using Prisma.
```
const toBeAddedArray=[
  {
    "title": "Lorem ipsum dolor",
    "url": "abc.com",
    "category": "business",
    "source": "627b99ca127e484a9c8cf96d"
  },
  {
    "title": "Lorem ipsum dolor sit amet",
    "url": "pqr.com",
    "category": "business",
    "source": "627b99ca127e484a9c8cf96d"
  },
.......
]
```
Here is the schema:
```
model posts {
  id        String   @id @default(auto()) @map("_id") @db.ObjectId
  url       String
  title     String
  category  String
  source    String   @db.ObjectId
}
```
Here is the prisma logics I tried:

Attempt 1.
```
const createManyPosts = await prisma.posts.createMany({
             toBeAddedArray 
            })
```

Error: Unknown arg `toBeAddedArray` in toBeAddedArray for type AffectedRowsOutput

Attempt 2.
```
const createManyPosts = await prisma.posts.createMany({
             data:{toBeAddedArray} 
            })
```
Error: Unknown arg `toBeAddedArray` in data.0.toBeAddedArray for type postsCreateManyInput.

Please guide me.

**Comment by antja03** [comment, marked as answer]
You want the `data` property to be the array of records to create. So rather than wrapping it in curly braces, your code should look like: 
```ts
const toBeAddedArray=[
  {
    "title": "Lorem ipsum dolor",
    "url": "abc.com",
    "category": "business",
    "source": "627b99ca127e484a9c8cf96d"
  },
  {
    "title": "Lorem ipsum dolor sit amet",
    "url": "pqr.com",
    "category": "business",
    "source": "627b99ca127e484a9c8cf96d"
  }, 
  //...
]

const result = await prisma.posts.createMany({
    data: toBeAddedArray 
})
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
